### DISCUSSION TITLE
what's the meaning of function json.dump()'s param : error_handler_t::ignore.

### DISCUSSION THREAD

**Discussion by chloro-pn** [nlohmann/json#2231] [Q&A] [answered]
Is it just copy the binary data to the json or just delete the invalid utf-8 bytes ?
like : 
```c++
json j;
std::string data(ptr, n); // binary data.
j["some_data"] = data;
return j.dump(-1, ' ', false, json::error_handler_t::ignore); // can i use this param to pass binary data?
```
thanks for the answer : )
---------------------------------------------------------------------------------------------
Through the test, I found that this method can't work : ( 
Maybe I can do this by converting strings to vector\<char\>, but I don't know if it will lead to inefficiency:
```c++
for(auto each : str) {
  j["binary"].push_back(each); // str.size() : 2KB ~ 4KB.
}

**Comment by nlohmann** [comment, marked as answer]
The `ignore` parameter copies invalid UTF-8 bytes verbatim to the output. Maybe the name is not too optimal - I will adjust the [documentation](https://nlohmann.github.io/json/doxygen/classnlohmann_1_1basic__json_acfc564f36da667b7d9109879d34dc36c.html#acfc564f36da667b7d9109879d34dc36c).

Update: #2232

**Comment by nlohmann** [comment]
What exactly do you try to accomplish? I am asking, because the library has support for binary types, but I would need to understand your usecase better.

**Reply by chloro-pn** [reply]
I want to transfer both utf-8 data and binary data using JSON format. and I found the right way : using bson interface : )

**Reply by nlohmann** [reply]
Cool. Do you need further assistance on this?

**Reply by chloro-pn** [reply]
No, the document is very clear. Thank's a lot !
